src/Microsoft.Performance.Toolkit.Engine.Tests/ProcessingOptionsResolverTests.cs

```diff
@@ -0,0 +1,254 @@
+                new[]
+                {
+                    new FileDataSource("test1" + Source123DataSource.Extension),
+                    new FileDataSource("test2" + Source123DataSource.Extension),
+                    new FileDataSource("test3" + Source123DataSource.Extension)
+                }, new DefaultProcessingMode());
+
+            var dsg123_2 = new DataSourceGroup(
+                new[]
+                {
+                    new FileDataSource("test4" + Source123DataSource.Extension)
+                }, new DefaultProcessingMode());
+
+            var dsg4 = new DataSourceGroup(
+                new[]
+                {
+                    new FileDataSource("test5" + Source4DataSource.Extension),
+                }, new DefaultProcessingMode());
+
+            var dsg5 = new DataSourceGroup(
+                new[]
+                {
+                    new FileDataSource("test6" + Source5DataSource.Extension),
+                }, new DefaultProcessingMode());
+
+            ProcessingSourceReference.TryCreateReference(typeof(Source123DataSource), out var psr123);
+            ProcessingSourceReference.TryCreateReference(typeof(Source4DataSource), out var psr4);
+            ProcessingSourceReference.TryCreateReference(typeof(Source5DataSource), out var psr5);
+
+            var processorOptions = new[]
+            {
+                new ProcessorOptions(
+                    new[]
+                    {
+                        new OptionInstance(
+                            new Option('s', "test1"),
+                            "arg1"),
+                    }),
+                new ProcessorOptions(
+                    new[]
+                    {
+                        new OptionInstance(
+                            new Option('t', "test2"),
+                            "arg2"),
+                    }),
+            };
+
+            IDictionary<Guid, ProcessorOptions> map = new Dictionary<Guid, ProcessorOptions>()
+            {
+                { psr123.Guid,  processorOptions[0]},
+                { psr4.Guid,    processorOptions[1]},
+                { psr5.Guid,    processorOptions[0]},
+            };
+
+            var sut = new ProcessingSourceOptionsResolver(map);
+
+            Assert.IsNotNull(sut, "Options Resolver is null");
+
+            Assert.AreEqual(processorOptions[0], sut.GetProcessorOptions(psr123.Guid, dsg123_1), "ProcessorOptions differ from expected");
+            Assert.AreEqual(processorOptions[0], sut.GetProcessorOptions(psr123.Guid, dsg123_2), "ProcessorOptions differ from expected");
+            Assert.AreEqual(processorOptions[1], sut.GetProcessorOptions(psr4.Guid, dsg4), "ProcessorOptions differ from expected");
+            Assert.AreEqual(processorOptions[0], sut.GetProcessorOptions(psr5.Guid, dsg5), "ProcessorOptions differ from expected");
+        }
+
+        [TestMethod]
+        [UnitTest]
+        public void CustomProcessingOptionsResolver()
```

Comment:
this, and the mock resolver, can be removed. You're essentially testing your implementation of a mock, which does not add value to the suite of unit tests, since the mock isn't used by consumers of the SDK.

Reply:
I think it _can_ be useful to have a mock options resolver which is already tested for future tests which might want to check specific behaviors for options resolvers, in case customers have issues with specific code paths.

Feel free to unresolve if you feel strongly about deleting this test case.

Reply:
I'd prefer getting rid of it. Unless the mock is both very complicated and used actually test collateral, there's no reason to have unit tests for it.